src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -382,6 +382,11 @@ public void OnStarting(Func<object, Task> callback, object state)
         {
             lock (_onStartingSync)
             {
+                if (HasResponseStarted)
+                {
+                    ThrowResponseAlreadyStartedException(nameof(OnStarting));
```

Comment:
Use an enum rather than a string; string version is additional calls, better to push it out to the infrequently called function. (related background info https://github.com/dotnet/coreclr/pull/6634#issuecomment-238325004)